app/scripts/modules/kubernetes/src/rawResource/controller/FiltersPubSub.ts

```diff
@@ -0,0 +1,29 @@
+import { IK8sResourcesFiltersState } from '../component/K8sResourcesFilters';
+
+export interface ISubcriber {
+  notify(message: IK8sResourcesFiltersState): void;
+}
```

Comment:
I'd probably drop this interface and put the functions directly in the `subscribers` array.

You can type it like so:

```
type ISubscriber = (message: IK8sResourcesFiltersState) => void;
```

Then when pushing to the list:
```
  public subscribe(subcriber: ISubscriber) {
    this.subcribers.push(subcriber);
    return () => this.subscribers = this.subscribers.filter(x => x !== subscriber);
```